docs/experiments/project-depends-on.md

```diff
@@ -1,40 +1,72 @@
+
+A dependency consists of three keys:
 
 * `namespace`: The package registry where the third party dependency is found
 * `package`: The name of the third party dependency as it appears in the lockfile
 * `version`: A semantic version range. Uses [Python packaging specifiers](https://packaging.pypa.io/en/latest/specifiers.html) which support almost all NPM operators, except `^`
 
+So a `r2c-internal-project-depends-on` key will either look like this:
+```yaml
+r2c-internal-project-depends-on:
+  namespace: ...
+  package: ...
+  version: ...
+```
+or this:
+```yaml
+r2c-internal-project-depends-on:
+  depends-on-either:
+    - namespace: ...
+      package: ...
+      version: ...
+    - namespace: ...
+      package: ...
+      version: ...
+    ...
+```
+
 ## Example
 
-Here’s an example `project-depends-on` rule that looks for a known vulnerable version of the AWS CLI from April 2017, but only reports the vulnerability if the `s3` module (where the vulnerability is located) is actually being used:
+Here’s an example `r2c-internal-project-depends-on` rule that looks for a known vulnerable version of the AWS CLI from April 2017, but only reports the vulnerability if the `s3` module (where the vulnerability is located) is actually being used:
 
 ```yaml
 rules:
 - id: vulnerable-awscli-apr-2017
-  patterns:
-    - pattern-either:
-      - pattern: boto3.resource('s3', ...)
-      - pattern: boto3.client('s3', ...)
-    - r2c-internal-project-depends-on:
-      - namespace: pypi
-        package: awscli
-        version: "<= 1.11.82"
+  pattern-either:
+  - pattern: boto3.resource('s3', ...)
+  - pattern: boto3.client('s3', ...)
+  r2c-internal-project-depends-on:
+    namespace: pypi
+    package: awscli
+    version: "<= 1.11.82"
   message: this version of awscli is subject to a directory traversal vulnerability in the s3 module
   languages: [python]
 ```
 
+## Findings
+
+Findings produced by rules with the `r2c-internal-project-depends-on` key can be of two types: _reachable_ and _nonreachable_.
+A _reachable_ finding is one with both a dependency match and a pattern match: a vulnerable dependency was found and 
+the vulnerable part of the dependency (according to the patterns in the rule) is used somewhere in code. An _unreachable_ 
+finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. 
+Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind,
+along with all matched dependencies, in the `extra` field of semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.
+
+A finding will only be considered reachable if the file containing the pattern match actually depends on the dependencies in the lockfile containing the dependency match. A file depends on a lockfile if it is the nearest lockfile going up the directory tree.
```

Comment:
Change to the formatting of this section (please, delete the other lines):
```suggestion
## Findings

Findings produced by rules with the `r2c-internal-project-depends-on` rule can be of two types: _reachable_ and _nonreachable_.

- A _reachable_ finding is one with both a dependency match and a pattern match: a vulnerable dependency was found and the vulnerable part of the dependency (according to the patterns in the rule) is used somewhere in the code.
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind, along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.

A finding is only considered reachable if the file containing the pattern match actually depends on the dependencies in the lockfile containing the dependency match. A file depends on a lockfile if it is the nearest lockfile going up the directory tree.
```

Reply:
On top of formatting suggestion, I have some ideas for the unreachable definition (I am not sure about the "Both kinds of findings specify their kind" - which kind? See my suggestion below, I hope it makes sense):

Original:
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the code that was pattern matched. Unreachable findings are reported as coming from the lockfile that was dependency matched. Both kinds of findings specify their kind, along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.

Suggesting:
- An _unreachable_ finding is one with only a dependency match. Reachable findings are reported as coming from the **code** that was pattern matched. Unreachable findings are reported as coming from the **lockfile** that was dependency matched. For both types of findings, Semgrep specifies whether they are unreachable or reachable along with all matched dependencies, in the `extra` field of Semgrep's JSON output, using the `dependency_match_only` and `dependency_matches` fields, respectively.